src/components/UserList.jsx

```diff
@@ -1,35 +1,81 @@
-import * as React from 'react'
-import { useListUsersQuery } from '../services/users'
-import { SimpleGrid, Box, Button } from '@chakra-ui/react'
+import React, { useState } from "react";
+import { SimpleGrid, useDisclosure, Spinner } from "@chakra-ui/react";
+import {
+  useListUsersQuery,
+  useDeleteUserMutation,
+  useGetUserQuery,
+} from "../services/users";
+
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+import Card from "./Card";
+import Drawer from "./Drawer";
+
+import "./UserList.css";
+
 export default function UserList() {
+  const { data, error, isLoading } = useListUsersQuery();
+  const [deleteUser] = useDeleteUserMutation();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [currentUser, setCurrentUser] = useState(0);
+  const { currentData } = useGetUserQuery(currentUser);
 
-  const { data, error, isLoading } = useListUsersQuery()
   // Individual hooks are also accessible under the generated endpoints:
 
   const invalidUser = {
-    avatar: "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/226.jpg",
+    avatar:
+      "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/226.jpg",
     email: "kelsey.wynns@emboldhealth.com",
     firstName: "George",
     id: "invalid-id",
-    lastName: "Bluth"
-  }
+    lastName: "Bluth",
+  };
+
+  const handleDeleteUser = async (userId) => {
+    const deleteuser = await deleteUser(userId);
+
+    if (deleteuser.error) {
+      toast(deleteuser.error.data.error);
```

Comment:
this looks and is working great!

an advanced way to handle this is to add redux middleware using `isRejectedWithValue`. You can watch for all api errors and show a toast that way. Some info here, https://redux-toolkit.js.org/rtk-query/usage/error-handling#handling-errors-at-a-macro-level